evalml/pipelines/components/transformers/feature_selection/feature_selector.py

```diff
@@ -32,19 +31,18 @@ def transform(self, X, y=None):
             ww.DataTable: Transformed X
         """
         X_ww = infer_feature_types(X)
-        X = _convert_woodwork_types_wrapper(X_ww.to_dataframe())
-        self.input_feature_names = list(X.columns.values)
+        self.input_feature_names = list(X_ww.columns.values)
 
         try:
             X_t = self._component_obj.transform(X)
         except AttributeError:
             raise MethodPropertyNotFoundError("Feature selector requires a transform method or a component_obj that implements transform")
 
-        X_dtypes = X.dtypes.to_dict()
+        X_dtypes = X_ww.dtypes.to_dict()
         selected_col_names = self.get_names()
         col_types = {key: X_dtypes[key] for key in selected_col_names}
-        features = pd.DataFrame(X_t, columns=selected_col_names, index=X.index).astype(col_types)
-        return _retain_custom_types_and_initalize_woodwork(X_ww, features)
+        features = pd.DataFrame(X_t, columns=selected_col_names, index=X_ww.index).astype(col_types)
+        return _retain_custom_types_and_initalize_woodwork(X_ww.ww.logical_types, features)
```

Comment:
I think we can potentially rewrite the feature selectors to do the column subset selection with woodwork but I'm not sure if that's something we need to do now.